.teamcity/src/main/kotlin/configurations/CompileAll.kt

```diff
@@ -1,11 +1,12 @@
 package configurations
 
+import configurations.CompileAll.BuildCacheType.PRODUCTION_BUILD_CACHE
 import model.CIBuildModel
 import model.Stage
 
-class CompileAll(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(stage = stage, init = {
-    id(buildTypeId(model))
-    name = "Compile All"
+class CompileAll(model: CIBuildModel, stage: Stage, buildCacheType: BuildCacheType = PRODUCTION_BUILD_CACHE) : BaseGradleBuildType(stage = stage, init = {
+    id(buildTypeId(model, buildCacheType))
+    name = "Compile All ${buildCacheType.nameSuffix}".trim()
```

Comment:
❓ I think there should be no space before the suffix, so that for the default case we don't get `Compile All `.

Reply:
I think `.trim()` should take care of it